File: LeetCode/Top Interview Questions/Array/Easy/two_sum_hari.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        values = dict()
        for i, elem  in enumerate(nums):
            compliment = target - elem
            if compliment in values:
                return [values[compliment],i]
            values[elem] = i
            
        return []

File: LeetCode/Top Interview Questions/Array/Easy/two_sum_hari.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>ans;
        unordered_map<int, int> hset;
        int i,l;
        l=nums.size();
        for(i=0;i<l;i++)
        {
            if(hset.find(target-nums[i]) != hset.end())
            {
                ans.push_back(hset[target-nums[i]]);
                ans.push_back(i);
                return ans;
            }
            else
            {
                hset[nums[i]]=i;
            }
        }
        return {};
        
    }
};
